### PR TITLE
[#2] Tailwind and PostCSS

### PR DESCRIPTION
**Pull request by joshuapease**
### Overview
- #2

Adds an empty Tailwind config and a standard PostCSS config.

Also including [`postcss-pxtorem`](https://www.npmjs.com/package/postcss-pxtorem) per @maxfenton's recommendation. 

It seems especially useful for arbitrary TW values (such as `m-[20px]`)

### How to test

Create a directory for your new Craft project

```bash
mkdir my-craft-project

# Then move into your directory
cd my-craft-project
```

Run the following `composer create-project` command. 

It uses this branch as the starter project.

```
docker run --rm -it -v "$PWD":/app -v ${COMPOSER_HOME:-$HOME/.composer}:/tmp composer create-project viget/craft-site-starter=dev-jp/vite-setup ./ --ignore-platform-reqs
```

After completing the setup steps (it prompts for your project's name), run the following commands.

👇 These will gradually become more automated as this project progresses.

```bash
# Start DDEV
ddev start

# Install Craft
ddev craft install

# Install Vite plugin
ddev craft plugin/install vite

# Launch the site!
ddev launch
```

Here's what you'll see!

❓ Also... somehow Vite is reloading when I edit .twig files. Even though I haven't added `vite-plugin-restart`

Anyone else want to confirm if I'm crazy or not?

CleanShot 2024-05-14 at 17 21 15@2x

### CODE REVIEW

**Comment by joshuapease** [comment]
@ekfuhrmann Wanted to get your thoughts on what a good starting point is for Tailwind.

An empty config? Or are there defaults that the UI team always reaches for?

File: src/css/app.css [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
+    @font-face {
+        font-display: swap;
+        font-family: 'Source Sans Pro';
+        font-style: normal;
+        font-weight: 700;
+        src: url('../fonts/sourcesanspro-bold-webfont.woff2') format('woff2');
+    }
 }
 
 body {
```
Comment (maxfenton) [inline src/css/app.css]: @joshuapease I think `body` and `strong` should also be wrapped in `@layer base { ... }`

Comment (joshuapease) [reply src/css/app.css]: Good catch! Resolved!

**Review by maxdmyers — APPROVED** [review]

**Review by maxfenton — APPROVED** [review]

File: tailwind.config.js
Comment (ekfuhrmann) [inline tailwind.config.js:1]: We've discussed this a bit in respects to **Blueprint** but the consensus was that we should include our own custom spacing system in the base tailwind. Not saying we should necessarily do it here, but having all of these values ready to go out of the box could be useful, and then it's easy enough to modify from there.

```js
spacing: {
  0: '0px',
  1: '1px',
  2: '2px',
  4: '4px',
  6: '6px',
  8: '8px',
  10: '10px',
  12: '12px',
  16: '16px',
  20: '20px',
  24: '24px',
  28: '28px',
  32: '32px',
  36: '36px',
  40: '40px',
  44: '44px',
  48: '48px',
  52: '52px',
  56: '56px',
  60: '60px',
  64: '64px',
  80: '80px',
  96: '96px',
  112: '112px',
  128: '128px',
},
```

**Comment by ekfuhrmann** [comment]
The only suggestion is to include our spacing system. Since we're using `postcss-pxtorem` (which I love), keying it up as just `px` values is fine. I normally using a helper function like `pxPair()` to simplify it a bit, but in the grand scheme of things is pretty unnecessary.

File: postcss.config.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                'right',
+                'inset',
+                'width',
+                'height',
+                'margin*',
+                'padding*',
+                'max-*',
+                'min-*',
+                'gap*',
+            ],
```
Comment (ekfuhrmann) [inline postcss.config.js:28]: Is this entirely necessary? We'll need to keep it upgraded based on props. We'll want to include `size-*` for instance.

Comment (maxfenton) [reply postcss.config.js:28]: This is necessary, yes. We can add `size-*`

Comment (maxfenton) [reply postcss.config.js:28]: This is just my running list, but if there are other CSS properties that might have pixel values we should add them.

Comment (maxfenton) [reply postcss.config.js:28]: Personally I hate using functions like `remPair` and `pxPair` and the like. I prefer using the values we get from designers in Figma and letting PostCSS handle the conversion. I mean, this is what I'd add to projects I'm on, but ultimately it’s up to UI, I guess.

Comment (maxfenton) [reply postcss.config.js:28]: @ekfuhrmann 
Ok, so we could use `propList: [*]` and not need to update this; as I was adding postcss-pxtorem to Viget projects, I wanted to be cautious and know what was being changed. 

I think we have two options: 
- an explicit list of properties (and add `size*` or `size-*`)
- a wildcard propList `[*]`

Comment (ekfuhrmann) [reply postcss.config.js:28]: Generally I use the following settings which seems to work best for me:

```
{
  rootValue: 16,
  propList: ['*'],
  mediaQuery: true,
  minPixelValue: 0,
},
```

Comment (joshuapease) [reply postcss.config.js:28]: Is `propList` referring to CSS properties? 

`size-` is just a Tailwind class right? 

I'm cool with using a `*` and letting projects opt out of specific properties. 

[From the docs](https://github.com/cuth/postcss-pxtorem?tab=readme-ov-file#options)

> Use ! to not match a property. Example: ['*', '!letter-spacing']

**Pull request merged**